src/components/interaction-designer/block-editors/PhoneRecorder.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <div v-if="isFeatureCallToRecordEnabled" class="phone-recorder">
+    <phone-recording-recorder-selector
+        v-model="callConfig"
+        @input="handleRecorderSelectionChanged"
+        :isModalVisible="isRecorderSelectorVisible"/>
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from 'vuex'
+import PhoneRecordingRecorderSelector from './PhoneRecordingRecorderSelector'
+
+export default {
+  components: {
+    PhoneRecordingRecorderSelector,
+  },
+
+  props: ['recordingKey'],
+
+  data() {
+    return {
+      // this is set by hooking into the v-model property of <phone-recording-recorder-selector v-model="callConfig">
+      // which $emits a new value that results in setting this variable by reference.
+      callConfig: null,
+    }
+  },
+
+  computed: {
+    ...mapState({
+      // todo: deprecate (this + setAudioRecordingConfigVisibilityForSelectedBlock) in favor of modal + local state
```

Comment:
@bzabos , this is an existing //TODO from voto5, shall we take the opportunity to deprecate them ? if yes, I may need more clarification on `in favor of modal + local state`